### ISSUE TITLE
Create a comparison tool for page transition 

### ISSUE THREAD

**Issue by enyo** [flutter/flutter#22862]
The default animation, when navigating between routes in a cupertino app doesn't look right.

1. it is way too slow
2. it seems as if it's completely linear without any acceleration curves

This results in a very bland and boring transition that doesn't feel native to iOS apps.

I know that I can define my own animations if I define the `CupertinoRoute` myself, but if I want to use the default router I didn't find a way to do that. I also think that out of the box, the animation should be as close to the native behaviour as possible.

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

**Comment by enyo** [comment]
```
  ╔════════════════════════════════════════════════════════════════════════════╗
  ║ A new version of Flutter is available!                                     ║
  ║                                                                            ║
  ║ To update to the latest version, run "flutter upgrade".                    ║
  ╚════════════════════════════════════════════════════════════════════════════╝


[✓] Flutter (Channel master, v0.9.7-pre.10, on Mac OS X 10.14 18A389, locale en-AT)
    • Flutter version 0.9.7-pre.10 at /Users/enyo/local/flutter
    • Framework revision b08f36e3cc (6 days ago), 2018-10-03 02:27:58 -0400
    • Engine revision 74662ab695
    • Dart version 2.1.0-dev.6.0.flutter-d2c5a24fd9

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.
      You may also want to add it to your PATH environment variable.


[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.18.0

[✓] Connected device (1 available)
    • iPhone XS Max • 8CC14555-4A03-4DF5-AED3-B0BB581ED902 • ios • iOS 12.0 (simulator)

! Doctor found issues in 1 category.
```

**Comment by xster** [comment]
I'm going to re-appropriate this issue.

Create a golden test tool like https://github.com/flutter/scroll_overlay where a native view and a Flutter view are overlaid and both are page transitioning at the same time.

- To help is adjust the curve if the next iOS version changes; and
- To produce videos showing the curve is the same with native.

**Comment by OscarFarrer** [comment]
I wrote an example comparing UINavigationController animation with CupertinoPageRoute. If you want I can share it to make it better.

I also made a video:

![screenvideo1](https://user-images.githubusercontent.com/47426255/52427703-bfd65580-2b00-11e9-8e76-24a8c40f1a59.gif)

**Comment by marcoms** [comment]
@OscarFarrer this looks very useful!

We actually recently updated the animation curve to try and improve things. Could you try this on the master/dev channel?

**Comment by OscarFarrer** [comment]
@marcoms How do I get the new animations? I've runned v.1.1.8 and nothing changed.

**Comment by OscarFarrer** [comment]
@marcoms How do I get the new animations? I've run v.1.1.8 and nothing changed.

**Comment by zoechi** [comment]
@OscarFarrer you need to upgrade Flutter `flutter channel master` `flutter upgrade`
```
[✓] Flutter (Channel master, v1.2.1-pre.125
```

**Comment by OscarFarrer** [comment]
Yes, in v1.2.1 there are new animations, the animations are faster and is good but when swiped is visible that CupertinoPageRoute animation ends first and then keeps stable while iOS still moves during the swipe. Another thing is that the screen behind in the transition still moves more than in iOS.

I made a new video with an iPad, I had to send it in 480p because it was too big for GitHub.
![screenvideoipad3](https://user-images.githubusercontent.com/47426255/52507383-acef7e00-2bf1-11e9-9b43-80edb3de2585.gif)

Maybe we should add a counter for transition progress.

**Comment by xster** [comment]
Thanks for running it again. I'd generally run in release mode since the Flutter version would end up doing a lot more runtime compilation than the native equivalent otherwise.

**Comment by OscarFarrer** [comment]
@xster I tried but flutter said it isn't possible with iOS simulator.
If you want I can share the code.

**Comment by xster** [comment]
Sure, that'd be very helpful

**Comment by OscarFarrer** [comment]
I tried another approach, I made a tool in Xcode to track page position during transition in a `while`; from the points I could get the animation curve that I found to be near Cubic(0.448,0.06, 0.599, 0.967). I think this could be a good method to find transition animation curve also for the future.

**Comment by marcoms** [comment]
Are you sure this is right? Looks to be [very similar](http://cubic-bezier.com/#.44,.06,.59,.96) to an ease-in-out curve

[This](http://cubic-bezier.com/#.35,.91,.33,.97) is what we previously came up with after studying a screen recording frame by frame

**Comment by OscarFarrer** [comment]
@marcoms Refining my tool I found though that the curve was more similar to that you said.

In a Cartesian plane the points are:

![screenshot](https://user-images.githubusercontent.com/47426255/52954014-4acd1080-3389-11e9-9b04-e702ccef984c.png)

Now we need to create the cubic from the points array.

This is the project of the tool, I didn't know how to share it and I sent the zip.
[AnimationTestApp.zip](https://github.com/flutter/flutter/files/2875779/AnimationTestApp.zip)

**Comment by OscarFarrer** [comment]
Do you think this kind of tool can replace the proposed transition comparison tool?

**Comment by OscarFarrer** [comment]
@xster @marcoms @enyo Any comments?

**Comment by xster** [comment]
Thanks, this is excellent. If you're willing, we'd like to merge it into a "platform fidelity comparison tools" repository. 

We'll likely want to augment it over time to to make sure the Flutter transition is overlaid or displayed side by side for comparison.

**Comment by xster** [comment]
@OscarFarrer, we made a repo https://github.com/flutter/platform_tests for these tools. If you'd like, you can send us a PR for it (though preferably let the iOS project be in the iOS directory of a `flutter create`d project so a side by side comparison can be made with Flutter as well as Android. 

Otherwise, you can just sign https://cla.developers.google.com/clas as well and we can move it for you.

**Comment by OscarFarrer** [comment]
@xster Done.

**Comment by marcoms** [comment]
After testing the fullscreen Cupertino page route on my app, I've concluded that the curve that is currently in use still falls short somewhat (the end is too abrupt).

Would be great to get some insights on this. @OscarFarrer can you share the raw data from your tool so we can compare it with the curve we use currently?

**Comment by Kavantix** [comment]
@marcoms lately I seemed to notice that the full page transition on iOS was too fast so I thought I would measure it.
From measuring the top of the page at every frame from a screen recording I got the following curves:
![Screenshot 2019-08-16 at 00 11 50](https://user-images.githubusercontent.com/6243755/63130477-e0ed9680-bfba-11e9-8744-0e5a24964129.png)
Notice here that the flutter curve ends a lot sooner than the native curve.
When only plotting the flutter curve I get this:
![Screenshot 2019-08-16 at 00 12 30](https://user-images.githubusercontent.com/6243755/63130508-fb277480-bfba-11e9-88d7-afb603f44cc6.png)
Which indeed looks quite similar to the native curve

I tested this on an iPad with the apps in slideover.
I'm also wondering if maybe the length of the animation is not fixed on native iOS since it took the native app 29 frames to finish the animation which is longer than the current 400ms that is used by the CupertinoPageRoute
